test/socket.jl

```diff
@@ -255,10 +255,14 @@ let
     end
     a, b, c = [create_socket() for i = 1:3]
     try
-        @sync begin
-            send(c, bcastdst, 2000, "hello")
-            recvs = [@async @test String(recv(s)) == "hello" for s in (a, b)]
-            map(wait, recvs)
+        # bsd family do not allow broadcasting to ip"255.255.255.255"
+        # or ip"127.255.255.255"
+        @static if !is_bsd() || is_apple()
+            @sync begin
+                send(c, bcastdst, 2000, "hello")
+                recvs = [@async @test String(recv(s)) == "hello" for s in (a, b)]
+                map(wait, recvs)
```

Comment:
We don't have a straightforward way to timeout on a `wait` call yet. One way to do this would be 

```
t0=time()
recvs_check = copy(recvs)
while (length(filter!(t->!istaskdone(t), recvs_check)) > 0) && (time() - t0 < TIMEOUT_VAL)
    sleep(0.05)
end
length(recvs_check) > 0 && error("timeout")
map(wait, recvs)
```